beacon-chain/sync/initial-sync/blocks_fetcher.go

```diff
@@ -363,9 +363,23 @@ func (f *blocksFetcher) requestBlocksByRoot(
 }
 
 // waitForBandwidth blocks up until peer's bandwidth is restored.
-func (f *blocksFetcher) waitForBandwidth(pid peer.ID) error {
+func (f *blocksFetcher) waitForBandwidth(pid peer.ID, count uint64) error {
 	log.WithField("peer", pid).Debug("Slowing down for rate limit")
-	timer := time.NewTimer(f.rateLimiter.TillEmpty(pid.String()))
+	rem := f.rateLimiter.Remaining(pid.String())
+	if uint64(rem) >= count {
+		// Exit early if we have sufficient capacity
+		return nil
+	}
+	// Determine how long it will take for us to have the
+	// required number of blocks allowed by our rate limiter.
+	// We do this by calculating the duration till the rate limiter
+	// can request these blocks without exceeding the provided
+	// bandwidth limits per peer.
+	tillEmpty := f.rateLimiter.TillEmpty(pid.String())
+	blocksNeeded := int64(count - uint64(rem))
+	currentNumBlks := f.rateLimiter.Capacity() - rem
+	expectedTime := int64(tillEmpty) * blocksNeeded / currentNumBlks
+	timer := time.NewTimer(time.Duration(expectedTime))
```

Comment:
Would be best if we abstract this into a small helper where the right math can be tested